core/core.go

```diff
@@ -262,7 +264,7 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, routingOptio
 
 	// setup exchange service
 	const alwaysSendToPeer = true // use YesManStrategy
-	bitswapNetwork := bsnet.NewFromIpfsHost(n.PeerHost, n.Routing)
+	bitswapNetwork := bsnet.NewFromIpfsHost(host, n.Routing)
```

Comment:
so that it can connect to nodes whose addresses may not have come from provider records. remember that provider records may come without addresses now that addresses expire quickly.